cmd/soroban-rpc/internal/feewindow/feewindow.go

```diff
@@ -0,0 +1,191 @@
+package feewindow
+
+import (
+	"io"
+	"slices"
+	"sync"
+
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+type FeeDistribution struct {
+	Max         uint64
+	Min         uint64
+	Mode        uint64
+	P10         uint64
+	P20         uint64
+	P30         uint64
+	P40         uint64
+	P50         uint64
+	P60         uint64
+	P70         uint64
+	P80         uint64
+	P90         uint64
+	P95         uint64
+	P99         uint64
+	FeeCount    uint32
+	LedgerCount uint32
+}
+
+type FeeWindow struct {
+	lock          sync.RWMutex
+	feesPerLedger *ledgerbucketwindow.LedgerBucketWindow[[]uint64]
```

Comment:
Should we decide now whether or not we ever want this to be DB-backed or if the in-memory approach is more than sufficient?

Reply:
In memory should be more than enough considering it’s 50 ledgers by default.